### ISSUE TITLE
Workers disappear from scheduling

### ISSUE THREAD

**Issue by beeva-carlosleon** [concourse/concourse#796]
# Bug Report
Hi,

We have experienced issues with the workers, that they disappear in command "fly -t whatever workers". We did not show any error traces in the worker logs, and the binary is active as a system service.

Rebooting worker works, but we need another solution more stable.

If you need an special log search to locate some kind of useful information, please let me know :)

Thanks,
Carlos León

The following can also be handy:

* Concourse version: 2.5.0 and previous version
* Deployment type (BOSH/Docker/binary): binary
* Browser (if applicable):

**Comment by databus23** [comment]
Sounds to me like a duplicate of #750

**Comment by beeva-carlosleon** [comment]
Today, we are facing the same issue:

The last traces in the logs are, before freece:

```
-- Logs begin at Mon 2016-11-21 09:43:28 EST. --
Nov 30 04:31:52 worker-3-concourse-eurofighter.localdomain concourse[5082]: {"timestamp":"1480498312.016620874","source":"guardian","message":"guardian.create.containerizer-create.watch.done","log_level":1,"data":{"handle":"4e5b72f9-1293-4e20-5c11-a68c2bfa9333","session":"251219.1.4"}}
Nov 30 04:31:52 worker-3-concourse-eurofighter.localdomain concourse[5082]: {"timestamp":"1480498312.116718054","source":"guardian","message":"guardian.create.containerizer-create.watch.done","log_level":1,"data":{"handle":"76e24468-7ec4-406c-63a6-2a44fdd22cef","session":"208620.1.4"}}
Nov 30 04:31:59 worker-3-concourse-eurofighter.localdomain concourse[5082]: {"timestamp":"1480498319.181214571","source":"baggageclaim","message":"baggageclaim.tick.reaping","log_level":1,"data":{"handle":"4fa66de8-2ee7-4e3e-4cc6-6d4b21af31d4","session":"41697","ttl":300}}
Nov 30 04:31:59 worker-3-concourse-eurofighter.localdomain concourse[5082]: {"timestamp":"1480498319.382294416","source":"baggageclaim","message":"baggageclaim.repository.destroy-volume.destroyed","log_level":1,"data":{"session":"3.3296131","volume":"4fa66de8-2ee7-4e3e-4cc6-6d4b21af31d4"}}
Nov 30 04:31:59 worker-3-concourse-eurofighter.localdomain concourse[5082]: {"timestamp":"1480498319.382365942","source":"baggageclaim","message":"baggageclaim.tick.reaping","log_level":1,"data":{"handle":"60951e2d-6e46-4a73-5c0c-92eeb6b2eee6","session":"41697","ttl":300}}
Nov 30 04:31:59 worker-3-concourse-eurofighter.localdomain concourse[5082]: {"timestamp":"1480498319.385970116","source":"baggageclaim","message":"baggageclaim.repository.destroy-volume.destroyed","log_level":1,"data":{"session":"3.3296132","volume":"60951e2d-6e46-4a73-5c0c-92eeb6b2eee6"}}
Nov 30 04:31:59 worker-3-concourse-eurofighter.localdomain concourse[5082]: {"timestamp":"1480498319.386009455","source":"baggageclaim","message":"baggageclaim.tick.reaping","log_level":1,"data":{"handle":"b2ca48da-c879-48a8-6fb2-6ca491ceb597","session":"41697","ttl":300}}
Nov 30 04:31:59 worker-3-concourse-eurofighter.localdomain concourse[5082]: {"timestamp":"1480498319.687027454","source":"baggageclaim","message":"baggageclaim.repository.destroy-volume.destroyed","log_level":1,"data":{"session":"3.3296133","volume":"b2ca48da-c879-48a8-6fb2-6ca491ceb597"}}
Nov 30 04:31:59 worker-3-concourse-eurofighter.localdomain concourse[5082]: {"timestamp":"1480498319.687118530","source":"baggageclaim","message":"baggageclaim.tick.reaping","log_level":1,"data":{"handle":"e2c9b300-23bd-40f2-4594-472af636f136","session":"41697","ttl":300}}
Nov 30 04:31:59 worker-3-concourse-eurofighter.localdomain concourse[5082]: {"timestamp":"1480498319.691161394","source":"baggageclaim","message":"baggageclaim.repository.destroy-volume.destroyed","log_level":1,"data":{"session":"3.3296134","volume":"e2c9b300-23bd-40f2-4594-472af636f136"}}

```
If you need more traces for investigate the issue, we can send more.

Thanks in advance,
Carlos León

**Comment by holstvoogd** [comment]
I'm running into some similar issues it seems. I have one host running ATC/TSA and two worker nodes. Last sunday worker1 got 'lost' and then all active builds on **worker2** got 'stuck'.

I looks like all checks are still executing on worker2, it just the builds that are stuck. The logs reflect that it is running the checks. Worker1 looks fine, but hasn't logged anything after the issue occurred first.

ATC is logging a lot of 'atc.baggage-collector.could-not-locate-worker' for worker1 every 30s, still going strong after ~24h of no worker1 ;)

Killing the build and starting it manually via the web interface works and seemed to fix the issue for some stuck builds. At least one completed and even triggered a following job. And the lost worker returns to services with a restart. So it all seemed fine at this point.

However, after stopping all stuck builds and restarting worker1 ATC still seems unable to schedule new builds; getting stuck *after* deciding all prerequisites are met to start the build. The restarted worker does get any containers assigned to it.

I then tried restarting the scheduler, which made things worse. ATC is now logging it has issues connecting to all kinds of ports I imagine are TSA related. A few log lines from ATC:
```json
{"timestamp":"Mon Dec 19 14:08:10 2016.395062447","source":"atc","message":"atc.syncer.docker-images:radar.scan.tick.init.image.check-image.init.retry-round-tripper.retrying","log_level":1,"data":{"error":"dial tcp 10.14.1.237:29585: getsockopt: connection refused","failed-attempts":6,"pipeline-scoped-name":"docker-images:resource:artifactory-resource-metadata","ran-for":"31.063543156s","session":"13.1.66.3.1.1.1.1.1"}}
{"timestamp":"Mon Dec 19 14:08:10 2016.401219130","source":"atc","message":"atc.syncer.docker-images:radar.scan.tick.init.retry-round-tripper.retrying","log_level":1,"data":{"error":"dial tcp 10.14.1.237:29585: getsockopt: connection refused","failed-attempts":6,"pipeline-scoped-name":"docker-images:resource:consul-proxy-repo","ran-for":"31.049639719s","session":"13.1.41.3.1.1"}}
{"timestamp":"Mon Dec 19 14:08:10 2016.401522875","source":"atc","message":"atc.syncer.docker-images:radar.scan.tick.init.retry-round-tripper.retrying","log_level":1,"data":{"error":"dial tcp 10.14.1.237:29585: getsockopt: connection refused","failed-attempts":6,"pipeline-scoped-name":"docker-images:resource:ssl-proxy-image-head","ran-for":"31.110724395s","session":"13.1.35.3.1.1"}}
```

Both workers are now reporting as active and are running checks it seems. But the build is still stuck on pending.

Only after restarting both workers and THEN restarting ATC it seems to have recovered.

I had this problem with 2.5.0 last week and now again with 2.5.1. The only thing relevant in the logs, was the following on from ATC:

```json
{"timestamp":"Sun Dec 18 14:39:52 2016.353486538","source":"atc","message":"atc.syncer.docker-images:radar.scan.tick.init.found-container-for-missing-worker","log_level":1,"data":{"container-handle":"4ab8ca3c-5943-4d89-52d8-2739677269e7","pipeline-scoped-name":"docker-images:resource:service-repo","session":"13.1.16.
12964.1","worker-name":"aws-concourse-worker1"}}
{"timestamp":"Sun Dec 18 14:39:52 2016.353841305","source":"atc","message":"atc.syncer.docker-images:radar.scan.tick.init.failed-to-look-for-existing-container","log_level":2,"data":{"error":"worker for container is missing","pipeline-scoped-name":"docker-images:resource:service-repo","session":"13.1.16.12964.1"}}
{"timestamp":"Sun Dec 18 14:39:52 2016.354040861","source":"atc","message":"atc.syncer.docker-images:radar.scan.tick.failed-to-initialize-new-resource","log_level":2,"data":{"error":"worker for container is missing","pipeline-scoped-name":"docker-images:resource:service-repo","session":"13.1.16.12964"}}
{"timestamp":"Sun Dec 18 14:39:52 2016.354469538","source":"atc","message":"atc.syncer.docker-images:radar.scanner-failed","log_level":2,"data":{"error":"worker for container is missing","member":"docker-images:resource:service-repo","session":"13.1"}}
```

Beyond that, none of the log files seem indicate anything on what went wrong, nor are there any interesting messages in syslog/message/etc.

**Comment by vito** [comment]
Consolidating this into https://github.com/concourse/concourse/issues/750
